### DISCUSSION TITLE
Discrepancy in AsyncScheduler API Documentation for add_schedule Method

### DISCUSSION THREAD

**Discussion by lixing-lx** [agronholm/apscheduler#1008] [Q&A]
Why does the API reference for the documentation show that the AsyncScheduler class's add_schedule method has a job_result_expiration_time parameter, but I can't find it in actual use? The version I am using is 4.0.0a5
![2024-12-28_164207_831](https://github.com/user-attachments/assets/803d5a13-ee2b-4059-b2d4-1124d954a016)

**Comment by HomerusJa** [comment]
The job_result_expiration_time parameter is something which was introduced after the release of v4.0.0a5, as it can be seen in the [versionhistory](https://apscheduler.readthedocs.io/en/master/versionhistory.html):

- BREAKING Added the job_result_expiration_time field to the Scheduleclass, to allow the job results from scheduled jobs to stay around for some time ([#927](https://github.com/agronholm/apscheduler/issues/927))

As there is (as far as I can tell) no documentation for the specific version 4.0.0a5, but rather only v3.x (with latest, stable and 3.x seemingly having the same content) and the master branch, this inconsistency persists. If you want to use the latest version available, you can install the package directly from the repository:

ˋˋˋ
pip install git+https://github.com/agronholm/apscheduler.git@master
ˋˋˋ

Notice that this version can change without your notice and include breaking changes, so you’d probably want to pin the commit using the following command:

ˋˋˋ
pip install git+https://github.com/agronholm/apscheduler.git@844dc05
ˋˋˋ

The commit used in this example is the latest commit at the time of writing this. This should not be used in production.

**Reply by lixing-lx** [reply]
Thank you very much. Your response solved my issue. Many of the unreleased features from the version installed directly from the repository make APScheduler much easier to use. I am using version 4, where the scheduler and worker are deployed in separate containers. The ease of use is far superior to Celery and APScheduler 3.x. I hope the author will release the official v4.0 version soon.

**Reply by HomerusJa** [reply]
Yea, I do so too. He mentioned recently to me that currently, v4.0 is feature-complete and it is just about finding and fixing bugs now.
